### PR TITLE
Update promscale K8S manifests links to 0.12.1

### PR DESCRIPTION
**Pull request by alejandrodnm**
# Description

The links for the manifests were outdated. This PR updates them for the promscale version 0.12.1 which was released on June 29th

# Links

Fixes #[insert issue link, if any]

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Are procedure and highlight tags used appropriately?
- [ ] Has the index been updated appropriately?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] Are all links provided in reference style, and resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: promscale/installation/kubernetes.md [hunk 1]
```diff
@@ -150,15 +150,15 @@ can provide the database URI, or specify connection parameters.
 ## Install Promscale with a manifest file
 This section includes instructions to install the Promscale Connector using a
 manifest file. To deploy TimescaleDB on Kubernetes use
-[helm charts][timescaledb-install-helm] instead. Alternatively, you can
-[install TimescaleDB on a host][timescaledb-host-install].
+[helm charts](#installing-the-timescaledb-helm-chart) instead. Alternatively, you can
+[install TimescaleDB on a host](#install-timescaledb).
```
Comment (Loquacity) [inline promscale/installation/kubernetes.md:154]: Why did you change these?

Comment (alejandrodnm) [reply promscale/installation/kubernetes.md:154]: Because for some reason those links were broken https://github.com/timescale/docs/runs/7135066225?check_suite_focus=true 

I went to the official docs [here](https://docs.timescale.com/promscale/latest/installation/kubernetes/#install-promscale-with-helm) to get the correct anchors to make the Github action pass. 

Do you want me to revert these 2 changes?

Comment (Loquacity) [reply promscale/installation/kubernetes.md:154]: No, it's all good! I was just curious, but it seems that doing anchors as reference style isn't working.

Comment (alejandrodnm) [reply promscale/installation/kubernetes.md:154]: I'm going a little blind with this. In the `latest` branch I get a 404 and without fixing those 2 links which I really don't know where they are pointing to:

I switched them to anchors because I thought it would work, because it's what works at https://docs.timescale.com/promscale/latest/installation/kubernetes/#installing-the-timescaledb-helm-chart clearly that didn't worked very good 😓 

Can you help me set up the correct links for this page @Loquacity ? 🥺

Comment (Loquacity) [reply promscale/installation/kubernetes.md:154]: Yeah, the way anchors work in our build system is a little janky, I'm sorry. I'll check this out and take a look for you.

Comment (Loquacity) [reply promscale/installation/kubernetes.md:154]: Both those links seem to be working fine in the live site as they are, can you confirm? I'm not sure we need to change anything at all here.

Comment (alejandrodnm) [reply promscale/installation/kubernetes.md]: Yeah, the live site for me it works correct. The issue was the CI on this PR, and the problem was with existing code, not with my changes. 

I just updated the branch. With your changes is all green now, so I'll request a re-review. 

Thanks so much for your help :)

**Review by Loquacity — APPROVED** [review]
